checkov/common/util/config_utils.py

```diff
@@ -0,0 +1,24 @@
+import os
+
+from pathlib import Path
+
+
+def config_file_paths(dir_path):
+    return [os.path.join(dir_path, '.checkov.yaml'), os.path.join(dir_path, '.checkov.yml')]
+
+
+def get_default_config_paths(argv):
+    """
+    Checkov looks for .checkov.yml or .checkov.yaml file in the directory (--directory) against which it is run.
+    If that does not have the config file, the current working directory is checked followed by checking the user's
+    home directory is searched.
+    :param argv: List of CLI args from sys.argv.
+    :return: List of default config file paths.
+    """
+    home_paths = config_file_paths(Path.home())
```

Comment:
@kartikp10 Can we parse this **after** we parsed the directory arg using configParse, instead of knitpicking through argv by ourselves?

Reply:
Hey @nimrodkor, that was my thought initially as well. The downsides of adding new "default_config_files" after the parser has been initialized are as follows:
- _default_config_files is a protected attribute ( not a blocker, we can define a setter in the extended class )
-  the --show-config command will not be able to print accurate information if the user runs it like `checkov -d terragoat --show-config` . For example:
```
# existing design. default paths are defined when the parser is initialized
❯ ckv --show-config -d terragoat
Command Line Args:   --show-config -d terragoat
Config File (terragoat/.checkov.yml):
  branch:            master
  check:             ['CKV_AWS_144']
  download-external-modules:False
  evaluate-variables:True
  external-modules-download-path:.external_modules
  framework:         all
  output:            cli

# if the default paths are added after arguments are parsed 
❯ ckv --show-config -d terragoat
Command Line Args:   --show-config -d terragoat
Defaults:
  --output:          cli
  --framework:       all
  --branch:          master
  --download-external-modules:False
  --external-modules-download-path:.external_modules
  --evaluate-variables:True
```
This is why I'd like to define default paths before the parser is initialized. Let me know if this makes sense.

Reply:
@kartikp10 I'm not sure I follow... Won't it include simply switching the order in `main.py` of lines 47 & 48? I think I'm still missing something, let's talk 🙂